expression/builtin.go

```diff
@@ -221,6 +221,19 @@ func (b *baseBuiltinFunc) getCtx() sessionctx.Context {
 	return b.ctx
 }
 
+func (b *baseBuiltinFunc) Clone() builtinFunc {
```

Comment:
Can we return baseBuiltinFunc rather than builtinFunc here?

Reply:
This is to implement the interface of `builtinFunc` interface.
Or we can do like this:
```
func (b *baseBuiltinFunc) clone() baseBuiltinFunc {
     do things like Clone() here.
}

func (b *baseBuiltinFunc) Clone() BuiltinFunc {
    make it panic or return b.clone()
}
```

Reply:
I prefer this way, and we can avoid many type assertions

Reply:
Now that every object inherit that `baseBuiltinFunc` have to rewrite `Clone()` function, why not just panic here?